### ISSUE TITLE
Prevent entering numeric page aliases

### ISSUE THREAD

**Issue by leofeyer** [contao/contao#1015]
**Affected version(s)**

4.8

**Description**  

While fixing contao/contao#744, I noticed that it is possible to enter numeric aliases such as `2` or `2/foo` (with folder URLs enabled), even though these aliases will not work because of contao/core-bundle#729. Even worse, when using the alias `2/foo`, `2` will be in the routing candidates, potentially causing unexpected results.

**How to fix**  

IMHO the most reasonable fix would be to trigger an error if someone enters a numeric alias. @contao/developers WDYT?

**Comment by Toflar** [comment]
Cannot really help here, I guess @aschempp has to because it concerns routing.

**Comment by leofeyer** [comment]
As briefly discussed with @aschempp, we should rediscuss whether we want to allow numeric aliases or not.

**Comment by leofeyer** [comment]
As discussed in Mumble on December 19th, we want to allow numeric aliases in the future, however, if there is a page with the ID 404 and no alias and the user tries to add the alias "404" to a new page, Contao must throw a "duplicate alias, please add an alias to the page ID 404 to use the alias 404" error message.

**Comment by gmpf** [comment]
I've just come across this issue in the wild and it sent me down a bit of a rabbit hole.

4.4.50 and 4.9.5 are also affected by the bug described in the initial message.

I've tried to follow the decisions in this thread, but I'm confused as to why the fix suggested by @leofeyer in the initial message ("trigger an error if someone enters a numeric alias") was not adopted until Contao can work with numeric aliases. Is it because the issue was never technically confirmed for an LTS?

Side note: Interestingly, an alias like `1234e0` also causes the problem because it `is_numeric`. The way Contao checks for IDs vs aliases was fixed in d3839c398db358d029ea8ca0f3ec7308d25cf043, but the Symfony CMF Router integration that was developed in parallel was not adjusted accordingly, and so the check there is still `is_numeric`, e. g.:  https://github.com/contao/contao/blob/f1530fc6b67f8a2d104943d9d0397a8a23dd5616/core-bundle/src/Routing/RouteProvider.php#L475
